mp/src/game/client/c_baseentity.cpp

```diff
@@ -1732,6 +1732,19 @@ void C_BaseEntity::SetNetworkOrigin( const Vector& org )
 	m_vecNetworkOrigin = org;
 }
 
+#ifdef NEO
+//-----------------------------------------------------------------------------
+// Purpose: When players become ragdolls their positions are not networked. This 
+// allows calculating the client side velocity of ragdolls for motion vision
+// Input  : org - old origin
```

Comment:
Are you suggesting I either
- pass GetRenderOrigin() instead of GetAbsOrigin()
- pass nothing and set value of m_vecOldOrigin to GetRenderOrigin()
- something else?

Reply:
This question was coming from a rather uneducated perspective of just eyeing the code without fully understanding the implications. But basically I was wondering it, instead of manually tracking the new/old velocities and all of that jazz to get the player position agnostic of whether the player had ragdolled could be simplified into calling this, to bypass the: `Purpose: When players become ragdolls their positions are not networked.` limitation that is mentioned as the reasoning for this code.

But if you think the answer is "no, we can't reasonably do this" then that's fine. Just figured I'd throw the idea out there, as I really don't know. Just in case you hadn't come across this helper method.

I.e. feel free to close this one as resolved, unless you feel there's some obvious easy refactoring wins here to be had.

Reply:
I was inspecting the value of (this) in the draw function above where we are calculating whether the entity is moving to apply the motion material override. None of the variables there were particularly helpful. m_pRagdoll has a velocity variable which looked promising but I fairly certain after some testing that that's just the initial velocity of the ragdol when it is spawned and isn't updated thereafter.

I'm interested in the velocity here, which is why I am storing the old origin and the time when the old origin was stored to calculate it myself, I don't think GetRenderOrigin provides me with an easier way of getting that velocity